rules/CodeQuality/Rector/Empty_/SimplifyEmptyCheckOnEmptyArrayRector.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\CodeQuality\Rector\Empty_;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\Array_;
+use PhpParser\Node\Expr\BinaryOp\Identical;
+use PhpParser\Node\Expr\Empty_;
+use PHPStan\Type\ArrayType;
+use Rector\Core\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\Tests\CodeQuality\Rector\Empty_\SimplifyEmptyCheckOnEmptyArrayRectorTest\SimplifyEmptyCheckOnEmptyArrayRectorTest
+ */
+final class SimplifyEmptyCheckOnEmptyArrayRector extends AbstractRector
+{
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Simplify `empty` functions calls on empty arrays',
+            [new CodeSample('$array = []; if(empty($values))', '$array = []; if([] === $values)')]
+        );
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [Empty_::class];
+    }
+
+    /**
+     * @param Empty_ $node
+     */
+    public function refactor(Node $node): ?Node
+    {
```

Comment:
`$node->expr` should be only Variable, PropertyFetch, StaticPropertyFetch for starter to avoid regression on complex expression

Reply:
Added check for on instance type off expression.

This did mean i had to remove case:
```
    public function functionCallInsideEmpty($values): bool
    {
        return is_array($values) && empty(array_filter($values));
    }
```

As that is a function call.